WarehouseManagement.Core/Services/EntryService.cs

```diff
@@ -0,0 +1,223 @@
+using WarehouseManagement.Common.Statuses;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs.Entry;
+using WarehouseManagement.Infrastructure.Data.Common;
+using WarehouseManagement.Infrastructure.Data.Models;
+using static WarehouseManagement.Common.MessageConstants.Keys.EntryMessageKey;
+using static WarehouseManagement.Common.MessageConstants.Keys.ZoneMessageKeys;
+
+namespace WarehouseManagement.Core.Services;
+
+public class EntryService : IEntryService
+{
+    private readonly IRepository repository;
+
+    public EntryService(IRepository repository)
+    {
+        this.repository = repository;
+    }
+
+    public async Task CreateAsync(EntryFormDto model, string userId)
+    {
+        var entry = new Entry()
+        {
+            Pallets = model.Pallets,
+            Packages = model.Packages,
+            Pieces = model.Pieces,
+            CreatedAt = DateTime.UtcNow,
+            CreatedByUserId = userId
+        };
+
+        await repository.AddAsync(entry);
+        await repository.SaveChangesAsync();
+    }
+
+    public async Task DeleteAsync(int id, string userId)
+    {
+        if (!await ExistsByIdAsync(id))
+        {
+            throw new KeyNotFoundException(EntryWithIdNotFound);
+        }
+
+        var entry = (await repository.GetByIdAsync<Entry>(id))!;
+
+        repository.SoftDelete(entry);
+        await repository.SaveChangesAsync();
+    }
+
+    public async Task EditAsync(int id, EntryFormDto model, string userId)
+    {
+        if (!await ExistsByIdAsync(id))
+        {
+            throw new KeyNotFoundException(EntryWithIdNotFound);
+        }
+
+        var entry = (await repository.GetByIdAsync<Entry>(id))!;
+
+        entry.Pallets = model.Pallets;
+        entry.Packages = model.Packages;
+        entry.Pieces = model.Pieces;
+
+        await repository.SaveChangesWithLogAsync();
+    }
+
+    public async Task<bool> ExistsByIdAsync(int id)
+    {
+        return await repository.GetByIdAsync<Entry>(id) != null;
+    }
+
+    public async Task<IEnumerable<EntryDto>> GetAllAsync(int? zoneId, ZoneEntryStatuses[]? statuses)
+    {
+        var query = repository.AllReadOnly<Entry>();
+
+        if (zoneId != null)
+        {
+            query = query.Where(e => e.ZoneId == zoneId);
+        }
+
+        if (statuses != null)
+        {
+            if (statuses.Contains(ZoneEntryStatuses.Waiting))
+            {
+                query = query.Where(e =>
+                    e.StartedProccessing == null && e.FinishedProccessing == null
+                );
+            }
+
+            if (statuses.Contains(ZoneEntryStatuses.Waiting))
+            {
+                query = query.Where(e =>
+                    e.StartedProccessing != null && e.FinishedProccessing == null
+                );
+            }
+
+            if (statuses.Contains(ZoneEntryStatuses.Waiting))
+            {
+                query = query.Where(e => e.FinishedProccessing != null);
+            }
```

Comment:
you are having repeated code here, in `GetAllWithDeletedAsync` and in `ZonesController`.
Either do a util or a class or maybe look into the factory pattern